tests/test_download.py

```diff
@@ -220,3 +222,117 @@ def test_download_multiple_files(client, publisher_wallet, consumer_wallet, web3
     download_endpoint = BaseURLs.SERVICES_URL + "/download"
     response = client.get(download_endpoint, query_string=payload)
     assert response.status_code == 200, f"{response.data}"
+
+
+@pytest.mark.integration
+def test_download_compute_asset_by_c2d(client, publisher_wallet, consumer_wallet, web3):
+    asset = get_dataset_ddo_with_multiple_files(client, publisher_wallet)
+    service = get_first_service_by_type(asset, ServiceType.ACCESS)
+
+    mint_100_datatokens(
+        web3, service.datatoken_address, consumer_wallet.address, publisher_wallet
+    )
+
+    tx_id, _ = start_order(
+        web3,
+        service.datatoken_address,
+        consumer_wallet.address,
+        service.index,
+        get_provider_fees(asset.did, service, consumer_wallet.address, 0),
+        consumer_wallet,
+    )
+
+    nonce = str(datetime.utcnow().timestamp())
+    _msg = f"{asset.did}{nonce}"
+
+    # Consume using url index and auth token
+    # (let the provider do the decryption)
```

Comment:
I think this comment is incorrect. There's no auth token here. And when you say `(let the provider do the decryption)` I think you mean, `(let the provider fetch on-chain ddo using tx_id)`

The provider always does the decryption, whether it fetches the ddo from chain, or its provided in args.

Reply:
Or you could just remove the comment.